packages/nimble-components/src/rich-text/mention-list-box/index.ts

```diff
@@ -0,0 +1,324 @@
+
+    /**
+     * @internal
+     */
+    public activeCharacterChanged(): void {
+        this.setActiveMappingConfigs();
+    }
+
+    /**
+     * @internal
+     */
+    public anchorElementChanged(prev: HTMLElement, next: HTMLElement): void {
+        if (prev) {
+            this.intersectionObserver.unobserve(prev);
+        }
+        if (this.region && this.anchorElement) {
+            this.region.anchorElement = this.anchorElement;
+            this.region.update();
+            this.intersectionObserver.observe(next);
+        }
+    }
+
+    /**
+     * @internal
+     */
+    public childItemsChanged(_prev: ListOption[], _next: ListOption[]): void {
+        this.filterOptions();
+        void this.selectFirstOption();
+    }
+
+    /**
+     * @internal
+     */
+    public listBoxChanged(): void {
+        if (this.listBoxNotifier) {
+            this.listBoxNotifier.unsubscribe(this);
+        }
+        this.listBoxNotifier = Observable.getNotifier(
+            this.listBox
+        );
+        this.listBoxNotifier.subscribe(this);
+    }
+
+    /**
+     * @internal
+     */
+    public regionChanged(): void {
+        if (this.regionNotifier) {
+            this.regionNotifier.unsubscribe(this);
+        }
+        this.regionNotifier = Observable.getNotifier(
+            this.region
+        );
+        this.regionNotifier.subscribe(this);
+    }
+
+    /**
+     * @internal
+     */
+    public handleChange(source: unknown, args: unknown): void {
+        if (
+            source instanceof Listbox
+            && typeof args === 'string'
+        ) {
+            if (args === 'selectedIndex') {
+                this.scrollOptionIntoView();
+            }
+        }
+        if (source instanceof AnchoredRegion) {
+            if (args === 'initialLayoutComplete') {
+                this.scrollOptionIntoView();
+            }
+        }
+    }
+
+    private get firstSelectedOption(): ListOption | null {
+        return this.listBox.selectedOptions[0] as ListOption ?? null;
+    }
+
+    private get hasSelectableOptions(): boolean {
+        return this.childItems.length > 0 && !this.childItems.every(o => o.disabled);
+    }
+
+    private scrollOptionIntoView(): void {
+        requestAnimationFrame(() => {
+            this.firstSelectedOption?.scrollIntoView({ block: 'nearest' });
+        });
+    }
+
+    private setActiveMappingConfigs(): void {
+        this.activeMappingConfigs = this.activeCharacter
+            ? this.mentionExtensionConfig?.find(
+                config => config.character === this.activeCharacter
+            )?.mappingConfigs
+            : undefined;
+    }
+
+    private filterOptions(): void {
+        if (!this.childItems || this.filter === undefined) {
+            return;
+        }
+        const normalizedFilter = normalizeString(this.filter.trim());
+        this.childItems.forEach(listOption => {
+            const normalizedOptionText = normalizeString(listOption.text.trim());
```

Comment:
Don't need to trim `listOption.text`, because [it already does it.](https://github.com/microsoft/fast/blob/2ea80697bc3a5193e6123fb08ac3be2a76571aeb/packages/web-components/fast-foundation/src/listbox-option/listbox-option.ts#L204)

Reply:
Removed the trimming logic.